packages/core/src/integrations/mcp-server/piiFiltering.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * PII filtering for MCP server spans
+ * Removes sensitive data when sendDefaultPii is false
+ */
+
+import { MCP_TOOL_RESULT_CONTENT_ATTRIBUTE } from './attributes';
+
+/** PII attributes that should be removed when sendDefaultPii is false */
+const PII_ATTRIBUTES = new Set([
+  'client.address',
+  'client.port',
+  'mcp.logging.message',
+  'mcp.resource.uri',
+  MCP_TOOL_RESULT_CONTENT_ATTRIBUTE,
+]);
+
+/**
+ * Removes PII attributes from span data when sendDefaultPii is false
+ */
+export function filterMcpPiiFromSpanData(
+  spanData: Record<string, unknown>,
+  sendDefaultPii: boolean,
+): Record<string, unknown> {
+  if (sendDefaultPii) {
+    return spanData; // Send everything when PII is allowed
+  }
+
+  // Use Object.fromEntries with filter for a more functional approach
+  return Object.fromEntries(
+    Object.entries(spanData).filter(([key]) => {
```

Comment:
l: I would prefer if we called `reduce` on `Object.entries(spanData)` to do this operation instead of calling `filter` + `Object.fromEntries` on the constructed array.